### PR TITLE
Rework of the test.yml action

### PR DESCRIPTION
**Pull request by Zejnilovic**

### CODE REVIEW

**Comment by Zejnilovic** [comment]
This setup is:
1. Shorter
2. A lot faster per test run

**Review by yruslan — APPROVED** [review]

**Review by dk1844 — APPROVED** [review]
That, indeed, is pretty cool!

**Review by benedeki — APPROVED** [review]
So much simpler now.
Any idea if some of the combinations won't work together in the future?

**Comment by Zejnilovic** [comment]
> So much simpler now.
> Any idea if some of the combinations won't work together in the future?

This should be fine "indefinitely". If there is more, there is always an exclude option.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -7,36 +7,25 @@ on:
...
+            spark: 3.0.1
+    name: Test Spark ${{matrix.spark}} on Scala ${{matrix.scala}}
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
       - name: Setup Scala
         uses: olafurpg/setup-scala@v10
         with:
           java-version: "adopt@1.8"
+      - uses: actions/checkout@v2
```
Comment (yruslan) [inline .github/workflows/test.yml:28]: Might be a similar same leftover 5 lines above

Comment (Zejnilovic) [inline .github/workflows/test.yml]: this line is extra

**Pull request merged**